service/src/main/java/bio/terra/workspace/db/WorkspaceDao.java

```diff
@@ -151,44 +154,38 @@ public Workspace getWorkspace(UUID id) {
   }
 
   @WriteTransaction
-  public boolean updateWorkspace(UUID workspaceId, String name, String description) {
-    if (name == null && description == null) {
-      throw new MissingRequiredFieldException("Must specify name or description to update.");
+  public boolean updateWorkspace(
+      UUID workspaceId, String name, String description, Map<String, String> propertyMap) {
+    if (name == null && description == null && propertyMap == null) {
+      throw new MissingRequiredFieldException(
+          "Must specify name, description, or properties to update.");
     }
 
+    List<String> columnSet = new ArrayList<>();
     var params = new MapSqlParameterSource();
+    params.addValue("workspace_id", workspaceId.toString());
 
     if (name != null) {
       params.addValue("display_name", name);
+      columnSet.add(" display_name = :display_name");
     }
 
     if (description != null) {
       params.addValue("description", description);
+      columnSet.add(" description = :description");
     }
 
-    return updateWorkspaceColumns(workspaceId, params);
-  }
-
-  /**
-   * This is an open ended method for constructing the SQL update statement. To use it, build the
-   * parameter list making the param name equal to the column name you want to update. The method
-   * generates the column_name = :column_name list. It is an error if the params map is empty.
-   *
-   * @param workspaceId workspace identifier - not strictly necessarily, but an extra validation
-   * @param columnParams sql parameters
-   */
-  private boolean updateWorkspaceColumns(UUID workspaceId, MapSqlParameterSource columnParams) {
-    StringBuilder sb = new StringBuilder("UPDATE workspace SET ");
-
-    sb.append(DbUtils.setColumnsClause(columnParams));
```

Comment:
why not use DbUtils.setColumnsClause for adding the params to the columnSet?

Reply:
You fix that allowed json properties wasn't in yet. I'll rebase and use that method.

Reply:
I think it is in https://github.com/DataBiosphere/terra-workspace-manager/blob/6ed5226ae3bbe21363a98b2159bc35d74b0666ad/service/src/main/java/bio/terra/workspace/db/DbUtils.java#L33